server.go

```diff
@@ -787,3 +793,15 @@ func (w *response) Close() error {
 	}
 	return nil
 }
+
+// ConnectionState() implements the ConnectionStater.ConnectionState() interface.
+func (w *response) ConnectionState() *tls.ConnectionState {
+	type tlsConnectionStater interface {
```

Comment:
I'm not 100% sure what @tmthrgd meant here, but this looks weird to me? Are we now back in the discussion on why we're returning a pointer to "connectionState"?